### PR TITLE
dts/nxp: Fix dtc v1.4.6 warnings on nxp boards/socs

### PR DESCRIPTION
**Pull request by MaureenHelm**
Fixes various new warnings seen when upgrading the device tree compiler to v1.4.6.

### CODE REVIEW

**Comment by MaureenHelm** [comment]
recheck

**Comment by mike-scott** [comment]
Fixes warnings for me.  

Tested-by Michael Scott <michael@opensourcefoundries.com>

File: boards/arm/colibri_imx7d_m4/colibri_imx7d_m4.dts [hunk 1]
```diff
@@ -30,17 +30,25 @@
 
 	leds {
 		compatible = "gpio-leds";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
 		green_led: led@0 {
 			gpios = <&gpio1 2 GPIO_INT_ACTIVE_LOW>;
 			label = "User LED1";
+			reg = <0>;
```
Comment (erwango) [inline boards/arm/colibri_imx7d_m4/colibri_imx7d_m4.dts]: FYI, for this particular warning type, I proposed to enable a waiver (cf https://github.com/zephyrproject-rtos/zephyr/issues/7564). Rational is I don't think adding a reg makes much sense here (while removing @0 would have bad side effects).

**Review by erwango — APPROVED** [review]
One comment (to open a discussion), otherwise LGTM

File: boards/arm/colibri_imx7d_m4/colibri_imx7d_m4.dts (hunk 1 above)
Comment (galak) [reply boards/arm/colibri_imx7d_m4/colibri_imx7d_m4.dts]: I agree w/Erwan, we shouldn't add reg here.

**Review by galak — CHANGES_REQUESTED** [review]
As @erwango mentioned, lets drop the adding of reg associated with that warning.

File: boards/arm/colibri_imx7d_m4/colibri_imx7d_m4.dts (hunk 1 above)
Comment (MaureenHelm) [reply boards/arm/colibri_imx7d_m4/colibri_imx7d_m4.dts]: Should I still add #address-cells and #size-cells? If I do, I get a different warning:
```
frdm_k64f.dts_compiled: Warning (avoid_unnecessary_addr_size): unnecessary #address-cells/#size-cells without "ranges" or child "reg" property in /leds
frdm_k64f.dts_compiled: Warning (avoid_unnecessary_addr_size): unnecessary #address-cells/#size-cells without "ranges" or child "reg" property in /gpio_keys
```

**Comment by MaureenHelm** [comment]
> I'm also seeing a warning regarding the typo of the usbd node address here: https://hastebin.com/uleqaqimeb.diff
> 
> Do you want to add a patch here fixing that?

I'm not seeing the dtc warning, but yes I can fix the typo.

**Comment by MaureenHelm** [comment]
Superseded by #7832

**Pull request closed without merging**